### PR TITLE
lunatask 2.0.12 (new cask)

### PR DESCRIPTION
**Pull request by mtsz-pl**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

**Comment by mateuszjablonski** [comment]
I suppose there is a way to create livecheck using https://lunatask.app/download/ but I'm not sure how to do that, any help/guidance is much appreciated - I can update the PR.

File: Casks/l/lunatask.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    regex(%r{<id>https://lunatask\.app/releases/(\d+(?:\.\d+)+)</id>}i)
+  end
+
+  depends_on macos: ">= :catalina"
+
+  app "Lunatask.app"
+
+  zap trash: [
+    "~/Library/Application Support/@lunatask",
+    "~/Library/Logs/@lunatask",
```
Comment (khipp) [inline Casks/l/lunatask.rb:21]:
```suggestion
    "~/Library/Application Support/@lunatask",
    "~/Library/Logs/@lunatask",
    "~/Library/Preferences/com.mikekreeki.tasks.plist",
    "~/Library/Saved Application State/com.mikekreeki.tasks.savedState",
```

File: Casks/l/lunatask.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  sha256 "9516f96a77b2213b2e43cb801a31ebf609f8a5887ff3a81d8dc735516c01052a"
+
+  url "https://lunatask.app/download/Lunatask-#{version}-universal.dmg"
+  name "Lunatask"
+  desc "Encrypted to-do list, habit tracker, journaling, life-tracking and notes app"
+  homepage "https://lunatask.app/"
+
+  livecheck do
+    url "https://lunatask.app/releases/atom.xml"
+    regex(%r{<id>https://lunatask\.app/releases/(\d+(?:\.\d+)+)</id>}i)
```
Comment (khipp) [inline Casks/l/lunatask.rb:12]:
```suggestion
    regex(%r{releases/(\d+(?:\.\d+)+)}i)
    strategy :xml do |xml, regex|
      xml.get_elements("//id").map do |item|
        match = item.text&.strip&.match(regex)
        next if match.blank?

        match[1]
      end
    end
```

**Review by khipp — COMMENTED** [review]
Thank you for your contribution, @mateuszjablonski!

I left a few suggestions below. Please also squash your commits to keep the history cleaner. Thanks!

**Comment by mateuszjablonski** [comment]
Thank you for suggestions @khipp I have updated the code

**Review by khipp — APPROVED** [review]
Thank you, @mateuszjablonski!

**Pull request merged**